code/modules/mod/mod_control.dm

```diff
@@ -475,6 +477,23 @@
 	wearer.update_spacesuit_hud_icon("0")
 	wearer = null
 
+/obj/item/mod/control/proc/generate_suit_mask()
+	var/covered_slots = NONE
+	var/parts = get_parts(all = TRUE)
+	for(var/obj/item/part as anything in parts)
+		if(!get_part_datum(part).sealed)
+			parts -= part
+			continue
+		covered_slots |= part.slot_flags
+	if(suit_masks["[covered_slots]"])
+		return suit_masks["[covered_slots]"]
+	var/icon/slot_mask = icon('icons/blanks/32x32.dmi', "nothing")
+	for(var/obj/item/part as anything in parts)
+		slot_mask.Blend(icon(part.worn_icon, part.icon_state), ICON_OVERLAY)
+	slot_mask.Blend("#fff", ICON_ADD)
+	suit_masks["[covered_slots]"] = slot_mask
```

Comment:
can we actually cache my slot flags? part icons might change
if so, why not cache globally?

Reply:
> can we actually cache my slot flags?

yeah why not
> part icons might change 

skin changes reset the cache
> if so, why not cache globally?

a lotta effort.....